### DISCUSSION TITLE
[Question][Hw1] Can I change function definition to recursive definition?

### DISCUSSION THREAD

**Discussion by yusungsim** [prosyslab-classroom/cs524-program-analysis#404] [Q&A]
While I was solving HW-1-1, questions 1 and 2, I found it very natural to define the functions `sigma` and `zip` as recursive function. 

However in the skeleton codes, the functions are defined as non-recursive (ie. defined only using `let`). 

Can I change this to recursive definition? (ie. using `let rec`)

**Comment by mekty2012** [comment]
The README says so.

> Notice that students are not permitted to change directory structures and types of the functions, but allowed to change `let` to `let rec` if needed.

**Comment by yusungsim** [comment]
Oops, sorry I didn't checked that carefully. Thanks!
